### ISSUE TITLE
Command line usage (`nvim -l` or `nvim --headless`) does not print into the standard output

### ISSUE THREAD

**Issue by wookayin** [neovim/neovim#27084]
### Problem

Nvim running in the head-less mode, e.g., `nvim -l <lua script>` or `nvim --headless -c <cmds>` that writes some output, actually does not write into stdout (fd=2).

T

### Steps to reproduce

For `nvim -l`:

```bash
# print to console
❯❯❯ echo 'print("hi")' | nvim --clean -i NONE -n -l -
hi

# try to redirect stdout, but nope
❯❯❯ echo 'print("hi")' | nvim --clean -i NONE -n -l - >/dev/null
hi
```

Same for `echomsg`:

```bash
❯❯❯ nvim --clean -i NONE -n --headless -c 'echom "echom"' -c 'qa'  > /dev/null
echom
```

### Expected behavior

Nvim should write to stdout, so Nvim's stdout can be piped.

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2121+g06aebfa99

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

macOS 14

### Terminal name/version

xterm

### $TERM environment variable

xterm-256color

### Installation

build from repo

**Comment by zeertzjq** [comment]
Related #24184

**Comment by wookayin** [comment]
Right, error messages (highlighted in `ErrorMsg`) should be written to stderr. Such as uncaught errors from Lua (E5108) or vimscript, `:echoerr`, `nvim_err_write[ln]`, to name a few.
